Sources/HaCWebsiteLib/WorkshopManager.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+import Dispatch
+import DotEnv
+
+public struct WorkshopManager {
+  /// The list of workshops as derived from the workshops directory last time it was updated
+  public private(set) static var workshops: [Workshop] = []
+
+  /// Used to avoid multiple git pulls being run simultaneously
+  private static let serialQueue = DispatchQueue(label: "com.hac.website.ws-pull-queue")
+
+  /**
+    Ensures that an up-to-date copy of the workshops repo is available locally
+    We don't want this function running many times concurrently, hence 'unsafe'
+  */
+  private static func unsafePull() {
```

Comment:
To reflect the corresponding `update` method, might `unsafeUpdate` be a better name?